mlir/lib/Quantum/Transforms/ConversionPatterns.cpp

```diff
@@ -109,26 +109,32 @@ struct DeviceOpPattern : public OpConversionPattern<DeviceOp> {
         auto specs = op.getSpecs(); // maybe {} for using the default backend device
         if (specs && !specs->empty()) {
             const auto args = specs.value();
-            StringRef qirName = "__quantum__rt__device"; // (int8_t *, int8_t *) -> void
+            StringRef qirName =
+                "__quantum__rt__device_init"; // (int8_t *, int8_t *, int8_t *) -> void
 
             Type intPtrType = LLVM::LLVMPointerType::get(IntegerType::get(ctx, 8));
```

Comment:
Nitpick, but slightly more precise.

```suggestion
            Type charPtrType = LLVM::LLVMPointerType::get(IntegerType::get(ctx, 8));
```